gobblin-yarn/src/test/java/gobblin/yarn/GobblinWorkUnitRunnerTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.yarn;
+
+import java.io.IOException;
+import java.net.URL;
+
+import org.apache.curator.test.TestingServer;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.apache.hadoop.yarn.util.ConverterUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Optional;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+
+/**
+ * Unit tests for {@link GobblinWorkUnitRunner}.
+ *
+ * @author ynli
+ */
+@Test(groups = { "gobblin.yarn" })
+public class GobblinWorkUnitRunnerTest {
+
+  private static final int TEST_ZK_PORT = 3183;
+
+  private TestingServer testingZKServer;
+
+  private GobblinWorkUnitRunner gobblinWorkUnitRunner;
+
+  private GobblinApplicationMaster gobblinApplicationMaster;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    this.testingZKServer = new TestingServer(TEST_ZK_PORT);
+
+    URL url = GobblinWorkUnitRunnerTest.class.getClassLoader().getResource(
+        GobblinWorkUnitRunnerTest.class.getSimpleName() + ".conf");
+    if (url == null) {
+      Assert.fail();
```

Comment:
Add a message to all the `Assert.fail()` invocation so that the thrown exception is more clear.